### ISSUE TITLE
[FURB] Path write

### ISSUE THREAD

**Issue by gbaian10** [astral-sh/ruff#18409]
## Case 1
```py
with open("hello.txt", "w", encoding="utf-8") as f:
    f.write("Hello, world!")
```

run `ruff check`

```console
PTH123 `open()` should be replaced by `Path.open()`
FURB103 `open` and `write` should be replaced by `Path("hello.txt").write_text("Hello, world!")`
```

## Case 2

If I try to fix `FURB103` first

```py
from pathlib import Path

Path("hello.txt").write_text("Hello, world!", encoding="utf-8")
```

run `ruff check`

```console
All checks passed!
```

## Case 3

But if we fix PTH123 first

```py
from pathlib import Path

with Path("hello.txt").open("w", encoding="utf-8") as f:
    f.write("Hello, world!")
```

run `ruff check`

```console
All checks passed!
```

---

Are Case3 and Case2 completely equivalent programs?
If they are, I would like to have a rule that can transform Case3 into Case2.

Similar to [unspecified-encoding (PLW1514)](https://docs.astral.sh/ruff/rules/unspecified-encoding/), will raise warnings for both `with open` and `Path.open`.

**Comment by MichaReiser** [comment]
I think it would make sense to extend `PTH123` to also support `Path::open`

**Comment by 11happy** [comment]
@MichaReiser I have linked the PR, &  I assume you meant FURB103  as it more justifed to extend that rather than PTH123 ?

> I think it would make sense to extend `PTH123` to also support `Path::open`
